internal/weldr/upload.go

```diff
@@ -291,9 +291,10 @@ func uploadRequestToTarget(u uploadRequest, imageType distro.ImageType) *target.
 			}
 		}
 
-		// The uploaded image object name must have 'tar.gz' suffix to be imported
+		// Providing the Object name is optional. If it is provided, we must
+		// ensure that it has a '.tar.gz' suffix to be successfully imported.
 		objectName := options.Object
-		if !strings.HasSuffix(objectName, ".tar.gz") {
+		if objectName != "" && !strings.HasSuffix(objectName, ".tar.gz") {
```

Comment:
I would suggest generating the object name here for the sake of consistency, see my other comment in the worker.

Reply:
Fine with me. Although if you insist, we should change it for AWS as well, because it generates the `key` already in the worker's code https://github.com/osbuild/osbuild-composer/blob/d9cfbea399426d4f52236c96a0cea769146919e4/cmd/osbuild-worker/jobimpl-osbuild.go#L469-L471